src/adapter/src/catalog/storage.rs

```diff
@@ -1514,6 +1552,54 @@ impl<'a> Transaction<'a> {
         }
     }
 
+    /// Updates role `id` in the transaction to `role`.
+    ///
+    /// Returns an error if `id` is not found.
+    ///
+    /// Runtime is linear with respect to the total number of items in the stash.
+    /// DO NOT call this function in a loop, use [`Self::update_roles`] instead.
+    pub fn update_role(&mut self, id: RoleId, role: SerializedRole) -> Result<(), Error> {
+        let n = self.roles.update(move |k, _v| {
+            if k.id == id {
+                Some(RoleValue { role: role.clone() })
+            } else {
+                None
+            }
+        })?;
+        assert!(n <= 1);
+        if n == 1 {
+            Ok(())
+        } else {
+            Err(SqlCatalogError::UnknownItem(id.to_string()).into())
+        }
+    }
+
+    /// Updates all roles with ids matching the keys of `roles` in the transaction, to the
+    /// corresponding value in `roles`.
+    ///
+    /// Returns an error if any id in `roles` is not found.
+    ///
+    /// NOTE: On error, there still may be some roles updated in the transaction. It is
+    /// up to the called to either abort the transaction or commit.
+    pub fn update_roles(&mut self, roles: BTreeMap<RoleId, SerializedRole>) -> Result<(), Error> {
```

Comment:
Are these functions called anywhere? Here for future use?

Reply:
`update_role` should be called for `ALTER ROLE`, but `update_roles` is never used. I'll delete `update_roles`, I just mindlessly added it.

Reply:
Removed.